modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/AbstractExtensionObjectFactory.java

```diff
@@ -55,13 +64,14 @@ public void setParameters(Map<String, Object> parameters)
      * Constructs a {@link ResolverSet} from the output of {@link #getParameters()},
      * using {@link #toValueResolver(Object)} to process the values.
      *
+     * @throws {@link ConfigurationException} if the exclusiveness condition between parameters is not honored
      * @return a {@link ResolverSet}
      */
-    protected ResolverSet getParametersAsResolverSet()
+    protected ResolverSet getParametersAsResolverSet(EnrichableModel model) throws ConfigurationException
     {
         ResolverSet resolverSet = new ResolverSet();
         getParameters().forEach((key, value) -> resolverSet.add(key, toValueResolver(value)));
-
+        checkParameterGroupExclusivenessForModel(model, resolverSet);
```

Comment:
this can be validated before creating the `resolverSet`, you need only the keyset from the parameters map.